src/main/scala/com/databricks/labs/overwatch/pipeline/Snapshot.scala

```diff
@@ -0,0 +1,365 @@
+
+    val cloneSpecs = buildCloneSpecs(cloneLevel,sourceToSnapFiltered)
+    val cloneReport = Helpers.parClone(cloneSpecs)
+    val cloneReportPath = s"${snapshotRootPath}/clone_report/"
+    cloneReport.toDS.write.format("delta").mode("append").save(cloneReportPath)
+    logger.log(Level.INFO, s"Clone report has been generated to this path: ${cloneReportPath}")
+  }
+
+}
+
+object Snapshot extends SparkSessionWrapper {
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+
+  def validate(sourceETLDB : String,
+               targetPrefix : String,
+               snapshotType : String,
+               pipeline : String = "Bronze,Silver,Gold",
+               tablesToExclude : String = " ",
+               cloneLevel: String = "Deep",
+               processType : String = "Snapshot"): Unit = {
+
+    // Check whether sourceETLDB is Overwatch Database
+    val isOverwatchDB = spark.sessionState.catalog.getDatabaseMetadata(sourceETLDB).properties.getOrElse("OVERWATCHDB", "FALSE").toBoolean
+    if (isOverwatchDB){
+      println(s"${sourceETLDB} is Overwatch Database and suitable for Snapshot")
+    }else{
+      val errMsg = s"${sourceETLDB} is Not Overwatch Database and not suitable for Snapshot"
+      throw new BadConfigException(errMsg)
+    }
+
+    // Snapshot Type should be Incremental or Full
+    if (snapshotType == "Incremental" || snapshotType == "Full"){
+      println(s"Snapshot Type is Suitable for Snapshot Process. Provided SnapshotType value is ${snapshotType}")
+    }else{
+      val errMsg = s"Provided SnapshotType value is ${snapshotType}. SnapshotType value should be either Full or Incremental. Can Not Proceed with Snapshot"
+      throw new BadConfigException(errMsg)
+    }
+
+    // Pipeline Should be Bronze, Sliver Or Gold
+
+    val pipelineList = pipeline.split(",").map(_.toLowerCase)
+
+    for (layer <- pipelineList){
+      if (layer == "bronze" || layer == "silver" || layer == "gold") {
+        println(s"Zone should be either Bronze,Silver or Gold. Provied Zone value is ${layer}")
+      }else
+      {
+        val errMsg = s"Unknown Zone found ${layer}, Zone should be either Bronze,Silver or Gold"
+        throw new BadConfigException(errMsg)
+      }}
+
+    // Clone Level should be "Deep" or "Shallow"
+    if (cloneLevel == "Deep" || cloneLevel == "SHALLOW"){
+      println(s"cloneLevel Type is Suitable for Snapshot Process. Provided cloneLevel value is ${cloneLevel}")
+    }else{
+      val errMsg = s"Provided cloneLevel value is ${cloneLevel}. cloneLevel value should be Deep Full or SHALLOW. Can Not Proceed with Snapshot"
+      throw new BadConfigException(errMsg)
+    }
+
+  }
+
+  def apply(
+             sourceETLDB: String,
+             targetPrefix : String,
+             snapshotType : String): Unit = {
+    apply(
+      sourceETLDB,
+      targetPrefix,
+      snapshotType,
+      pipeline = "Bronze,Silver,Gold",
+      tablesToExclude = " ",
```

Comment:
default for empty should not be one blank space, but instead, just ""

Reply:
Done